### PR TITLE
Add code for C_DeriveKey()

### PR DESCRIPTION
**Pull request by ruchi393**

### CODE REVIEW

File: libckteec/src/serialize_ck.c [hunk 1]
```diff
@@ -410,6 +410,70 @@ static CK_RV serialize_mecha_aes_iv(struct serializer *obj,
 	return serialize_buffer(obj, mecha->pParameter, mecha->ulParameterLen);
 }
 
+/*
+ *struct CK_KEY_DERIVATION_STRING_DATA {
+ *	CK_BYTE_PTR pData;
+ *	CK_ULONG    ulLen;
+ *};
+ */
```
Comment (etienne-lms) [inline libckteec/src/serialize_ck.c:418]: remove these lines

File: libckteec/src/serialize_ck.c [hunk 2]
```diff
@@ -410,6 +410,70 @@ static CK_RV serialize_mecha_aes_iv(struct serializer *obj,
...
+	size = sizeof(uint32_t) + param->ulLen;
+	rv = serialize_32b(obj, size);
+	if (rv)
+		return rv;
+
+	rv = serialize_ck_ulong(obj, param->ulLen);
+	if (rv)
+		return rv;
+
+	rv = serialize_buffer(obj, param->pData, param->ulLen);
```
Comment (etienne-lms) [inline libckteec/src/serialize_ck.c:439]: `return serialize_buffer(obj, param->pData, param->ulLen);`

Ditto in function below.

File: libckteec/include/pkcs11_ta.h [hunk 3]
```diff
@@ -565,6 +565,22 @@ enum pkcs11_ta_cmd {
 	 * This command relates to the PKCS#11 API function C_GenerateRandom().
 	 */
 	PKCS11_CMD_GENERATE_RANDOM = 42,
+
+	/*
+	 * PKCS11_CMD_DERIVE_KEY - Derive a key from parent key.
```
Comment (etienne-lms) [inline libckteec/include/pkcs11_ta.h:570]: "... from **a** parent key" (to sync with comment in https://github.com/OP-TEE/optee_os/pull/4397)

**Review by etienne-lms — COMMENTED** [review]
only minor comments. 
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` once addressed.

**Comment by ruchi393** [comment]
Comments addressed, tags applied.

File: libckteec/include/pkcs11_ta.h (hunk 3 above)
Comment (ruchi393) [reply libckteec/include/pkcs11_ta.h]: Done

File: libckteec/src/serialize_ck.c (hunk 2 above)
Comment (ruchi393) [reply libckteec/src/serialize_ck.c]: Changed at both the places

File: libckteec/src/serialize_ck.c (hunk 1 above)
Comment (ruchi393) [reply libckteec/src/serialize_ck.c]: Removed

**Pull request merged**